scripts/update-k8s.sh

```diff
@@ -38,8 +38,6 @@ while read -ra LINE; do
 done < <(curl -fsSL "https://raw.githubusercontent.com/kubernetes/kubernetes/$KUBERNETES_VERSION/go.mod" \
   | grep -E '^\s*k8s.io/\S+ v\S+$')
 
-deps+=("-replace k8s.io/kubernetes=k8s.io/kubernetes@$KUBERNETES_VERSION")
```

Comment:
Did you still use the (updated) script to bump the dependencies? I'm curious to know if we can just do `go get ...` directly and rely on Go modules / proper semver in Kubernetes repositories other than k/k to resolve the (transitive) dependencies correctly.

If that works, then I'd think it's preferable because we ideally shouldn't set `-replace` directives explicitly (as we still do in line 37); in that case, we could leave the script as-is in case we do need to revendor k/k again.

If the modified script is still needed and/or makes things easier, then let's keep it in the updated form by all means.

Reply:
I did use the script, but removing the `replace`s works fine, and without those there's really no need for the script. Removed the replaces and the script in additional commits.

Reply:
I also pondered writing a new script that would `go get` the right versions of things, but since `k8s.io/klog` and `k8s.io/utils` don't follow the same versioning pattern that's a little dicey. If I do this update again in the future I would just edit the go.mod file by hand and then `go mod tidy; go mod vendor`.